backend/backend.py

```diff
@@ -38,32 +39,32 @@ def start_video(self) -> None:
             video. """
 
     def on_base_canvas_updated(self, base_canvas: Canvas) -> None:
-        """ Catches any errors that occurs while the base canvas is being
+        """ Updates the base layer of the overlay whenever the canvas is updated,
+            which is shown on the camera.
+
+            Catches any errors that occurs while the base canvas is being
             updated. """
-        try:
+        with CameraException(self.client, self.backend_name):
```

Comment:
Another thing, rather than hardcoding `self.backend_name` I believe we can use `self.__class__.__name__` (see https://stackoverflow.com/questions/19335722/getting-name-of-subclass-from-superclass/19335909)